Feature/sdk

ChadRehmKineticData opened the pull request:
Due to the number of new handlers it would be easier to clone the repo to perform the review.

ChadRehmKineticData committed:
removed unneeded file

ChadRehmKineticData committed:
PER-245 initial commit of Custom Record Update
Soap based handler used to update custom fields on a custom record.

ChadRehmKineticData committed:
PER-245 Initial commit of Custom Record search
Updated the readme for the Customer Record update along with adding a Custom Record search handler.

ChadRehmKineticData committed:
PER-245 Initial commit of Custom Record retrieve

ChadRehmKineticData committed:
PER-245 initial commit of opportunity retrieve

ChadRehmKineticData committed:
PER-245 initial commit of Opportunity Search

ChadRehmKineticData committed:
PER-245 initial commit Custom Record Create
Also updated readme and simple input spelling typo.

btigges committed:
Corrected Opportunity Search Handler

ChadRehmKineticData committed:
PER-245 updated sdk search handlers
Added internal id to the search results.  Added columns to the custom record search

ChadRehmKineticData committed:
Merge branch 'feature/sdk' of github.com:kineticdata/peripherals-netsuite into feature/sdk

ChadRehmKineticData committed:
PER-245 updated opportunity retrieve handler
Removed unneeded parameter from node xml.

ChadRehmKineticData committed:
PER-245 updated the sdk search handlers
Added enable debug logging to logging statements.

ChadRehmKineticData committed:
PER-245 updated the sdk handlers

ChadRehmKineticData committed:
PER-245 small updates to handlers
Found small updates to make to handlers before PR review.

jackboespflug commented on handlers/netsuite_custom_record_create_v1/handler/dependencies.rb:
```diff
@@ -0,0 +1,254 @@
+  $:.unshift library_path
+  # Require the library
+  require 'builder'
+end
```
Instead of modifying the builder gem, we should just require the version file if the VERSION constant still isn't defined.

```
  # Require the library
  require 'builder'
  require 'builder/version' unless defined?(Builder::VERSION)
```

jackboespflug commented on handlers/netsuite_custom_record_create_v1/handler/vendor/GEM_MODIFICATIONS.txt:
```diff
@@ -0,0 +1,4 @@
+The following modifications were made to the vendor libraries in order to make
+them compatible with the Kinetic Task process:
+ * Added the VERSION constant to the Builder class in the
+   lib\builder\xmlbase.rb file at line 6.
```
We should remove the modification from the gem, and then this file should be deleted.

jackboespflug commented on handlers/netsuite_custom_record_create_v1/handler/vendor/builder-3.2.3/lib/builder/xmlbase.rb:
```diff
@@ -0,0 +1,200 @@
+require 'builder/blankslate'
+
+module Builder
+  VERSION = "3.2.3"
```
Per the GEM_MODIFICATIONS.TXT file, this line was manually added by us. We should put the gem back to it's original state and update the dependencies.rb file instead.

jackboespflug reviewed (changes requested):
Revert modifications to the builder gem for ALL handlers. I only outlined the changes in 1, but it should be done for all of them.

The readme.md and changelog.md files for the repository weren't updated with this PR, but there should probably be some info about these handlers in those files. Also probably indicate the version 2021_1 of the API is required.

ChadRehmKineticData committed:
PER-245 updated handlers as per PR requested changes
On top of making the requested changes also updates repo files.  Included new thumbnail image for git repo.

ChadRehmKineticData commented:
Made requested updates, but only tested one handler.  Since all updates were identical it is assumed that all other handlers will also function.
